contracts/src/CrossChainApplications/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -0,0 +1,192 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@subnet-evm-contracts/interfaces/IWarpMessenger.sol";
+import "@subnet-evm-contracts/interfaces/IAllowList.sol";
+import "@subnet-evm-contracts/interfaces/INativeMinter.sol";
+import "./INativeTokenDestination.sol";
+import "../../Teleporter/ITeleporterMessenger.sol";
+import "../../Teleporter/ITeleporterReceiver.sol";
+import "../../Teleporter/SafeERC20TransferFrom.sol";
+
+// Native Minter Precompile Contract Address
+address constant MINTER_ADDRESS = 0x0200000000000000000000000000000000000001;
+// This is the address where the burned base fee go. We use this to send the amount of
+// burned tokens back to the source chain.
+// TODO implement mechanism to report burned tx fees to source chian.
+address constant BURNED_TX_FEES_ADDRESS = 0x0100000000000000000000000000000000000000;
+// Designated Blackhole Address. This is where we burn token before sending an unlock
```

Comment:
Would you mind going through the comments and re-wording any places where "we" is used? I find as a general rule of thumb that it makes comments cleaner and clearer. "We" isn't well defined in the context of contracts deployed on-chain, and we don't want to portray ourselves as the owner of contracts in any way.

Reply:
nit: to add on there's a few "this is" or "this" uses in general can be made more clear